tests/unit/schema.test.js

```diff
@@ -0,0 +1,204 @@
+
+import {
+  validateCredentialSchema,
+} from '../../src/utils/vc';
+
+import exampleCredential from '../example-credential';
+import exampleSchema from '../example-schema';
+
+const exampleAuthor = 'did:dock:5CEdyZkZnALDdCAp7crTRiaCq6KViprTM6kHUQCD8X6VqGPW';
+
+describe('VerifiableCredential Tests', () => {
+  const vc = new VerifiableCredential(exampleCredential.id);
+  vc.addSubject(exampleCredential.credentialSubject);
+
+  test('VerifiableCredential\'s setSchema should appropriately set credentialSchema.', () => {
+    vc.setSchema(exampleCredential.credentialSchema.id, exampleCredential.credentialSchema.type);
+    expect(vc.credentialSchema).toMatchObject(
+      expect.objectContaining({
+        id: expect.anything(),
+        type: expect.anything(),
+      }),
+    );
+  });
+
+  test('VerifiableCredential\'s validateSchema should validate the credentialSubject with given JSON schema.', async () => {
+    await expect(vc.validateSchema(exampleSchema)).toBe(true);
+  });
+});
+
+describe('Basic Schema Tests', () => {
+  let keypair;
+  let schema;
+  beforeAll(async (done) => {
+    await cryptoWaitReady();
+    const keyring = new Keyring();
+    const seed = '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff';
+    keypair = keyring.addFromUri(randomAsHex(32), null, 'sr25519');
+
+    schema = new Schema();
+    schema.name = 'AlumniCredSchema';
+    schema.version = '1.0.0';
+
+    done();
+  }, 10000);
+
+  test('accepts the id optionally and generates id of correct size when id is not given', () => {
+    const schemaNoID = new Schema();
+    const encodedIDByteSize = 48;
+    expect(schemaNoID.id && schemaNoID.id.length).toBe(encodedIDByteSize + DockBlobQualifier.length);
+  });
+
+  test('setAuthor will set the author and accepts a DID identifier or full DID', () => {
+    schema.setAuthor(exampleAuthor);
+    expect(schema.author).toBe(exampleAuthor);
+  });
+
+  test('setJSONSchema will only accept valid JSON schema and set the schema key of the object.', async () => {
+    await expect(schema.setJSONSchema({
+      invalidSchema: true,
+    })).rejects.toThrow();
+
+    await schema.setJSONSchema(exampleSchema);
+    expect(schema.schema).toBe(exampleSchema);
+  });
+
+  test('validateSchema will check that the given schema is a valid JSON-schema.', async () => {
+    await expect(Schema.validateSchema(exampleSchema)).resolves.toBeDefined();
+  });
+
+  test('toJSON will generate a JSON that can be sent to chain.', () => {
+    const result = schema.toJSON();
+    console.log('toJSON:', result);
+    expect(result).toMatchObject(
+      expect.objectContaining({
+        id: expect.anything(),
+        schema: expect.anything(),
+      }),
+    );
+  });
+
+  test('toBlob will generate a JSON that can be sent to written with blob module', () => {
+    schema.setAuthor(createNewDockDID());
+    const result = schema.toBlob();
+    console.log('toBlob:', result);
+    expect(result).toMatchObject(
```

Comment:
It woulld be good to show that this blob can be written to chain.

Reply:
this is just a unit test, we have other tests for that in integration